### ISSUE TITLE
[BUG] SymNCO invariance loss minimizes cosine similarity

### ISSUE THREAD

**Issue by keep9oing** [ai4co/rl4co#283]
## Describe the bug

SymNCO's `invariance_loss` returns positive cosine similarity, which is added to the minimized training loss with a positive default weight (`alpha=0.2`). This favors dissimilar projected representations.

## To Reproduce

```python
import torch
from rl4co.models.zoo.symnco.losses import invariance_loss

aligned = torch.tensor([[[1., 0.]], [[1., 0.]]])
opposite = torch.tensor([[[1., 0.]], [[-1., 0.]]])
print(invariance_loss(aligned, 2))   # +1
print(invariance_loss(opposite, 2))  # -1
```

Run against upstream commit `7a1539678b6a81cc68b4069440aa5e98f6931f6c` before the fix. Aligned representations should have a lower loss.

## System info

RL4CO 0.6.0 (source checkout), Python 3.12.13, PyTorch 2.6.0+cpu, TorchRL 0.6.0, TensorDict 0.6.0, Linux.

## Reason and Possible fixes

Return `-similarity.mean()` in `rl4co/models/zoo/symnco/losses.py`.

## Additional context

A maintainer confirmed the sign issue and suggested submitting a fix.

## Checklist

- [x] I have checked that there is no similar issue in the repo.
- [x] I have provided a minimal working example to reproduce the bug.
